libs/core/src/CutFlowManager.cpp

```diff
@@ -19,10 +19,23 @@ CutFlowManager::CutFlowManager(shared_ptr<EventReader> eventReader_, shared_ptr<
     info() << "Weights branch not specified -- will assume weight is 1 for all events" << endl;
   }
 
-  if (eventReader->inputFile->Get("CutFlow")) {
-    info() << "Input file contains CutFlow directory - will store existing cutflow in the output." << endl;
+  RegisterPreExistingCuts();
+  for(auto collectionName : additionalCutFlowCollections){
+    RegisterCollection(collectionName);
+    RegisterPreExistingCuts(collectionName);
+  }
```

Comment:
Yeah that's how I use it now, but this way it can also be done the same way as it's done for the regular cutflow in the constructor if it's preferred. I could just remove it as well

Reply:
I'm not sure I understood what you tried to explain here ;) In the constructor I think the only thing we need to do is to check if there are per-collection cut flows in the input file, and if so - register them. I don't think we need the user to provide us with the list of per-collection cut flows at this stage. Then, if they try to update a cut flow and we realize it's not yet in the map - we should register it (because it means it was not in the input file and it's the first time they update it).

Reply:
Okey I changed this so that the cutflow manager looks for any pre-existing key that contains "CutFlow" so this works as long as all cutflows has this in it

Reply:
Now it's handled in RegisterPreExistingCuts, but for cutflows either named "CutFlow" or "CollectionCutFlow_"+collectionName in the root file as that's what we're saving the names as in the SaveCutFlow function. And if it's called something else I print a warning and the collectionName = cutflowName